schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -503,7 +507,37 @@ public Integer getSchemaVersion(String schemaName, String schemaText) throws Sch
         return result;
     }
 
-    private Integer findSchemaVersion(String type,
+    public SchemaVersionInfo registerSchema(String schemaName, String schemaText, String description)
+        throws UnsupportedSchemaTypeException, InvalidSchemaException, SchemaNotFoundException, IncompatibleSchemaException {
+        SchemaMetadataInfo schemaMetadataInfo = getSchemaMetadata(schemaName);
+        if (schemaMetadataInfo == null) {
+            SchemaMetadata schemaMetadata = new SchemaMetadata.Builder(schemaName).type(AvroSchemaProvider.TYPE).build();
```

Comment:
This class should not have any avro specific related stuff. Can you push this to `ConfluentSchemaRegistryCompatibleResource` as it is kind of specific to that?